test/sparse_merkle_tree.spec.js

```diff
@@ -46,6 +46,48 @@ describe('SparseMerkleTree', () => {
 		}
 	});
 
+    describe('remove keys from proof', () => {
+        for (const test of FixturesInclusionProof.testCases) {
+            it(test.description, async () => {
+				const smt = new SparseMerkleTree(32);
+				const inputKeys = test.input.keys;
+				const inputValues = test.input.values;
+				const queryKeys = test.input.queryKeys.map(keyHex => Buffer.from(keyHex, 'hex'));
+				const outputMerkleRoot = test.output.merkleRoot;
+
+				const kvpair = [];
+				for (let i = 0; i < inputKeys.length; i += 1) {
+					kvpair.push({ key: Buffer.from(inputKeys[i], 'hex'), value: Buffer.from(inputValues[i], 'hex') });
+				}
+
+				const rootHash = await smt.update(Buffer.alloc(0), kvpair);
+				expect(rootHash.toString('hex')).toEqual(outputMerkleRoot);
+
+				const proof = await smt.prove(rootHash, queryKeys);
+                await expect(smt.verifyInclusionProof(rootHash, queryKeys, proof)).resolves.toEqual(true);
+
+                const removedKeys = queryKeys.slice(0, queryKeys.length/2);
+                const newQueryKeys = queryKeys.slice(queryKeys.length/2, queryKeys.length);
+                const newProof = await smt.removeKeysFromProof(proof, removedKeys);
+                await expect(smt.verifyInclusionProof(rootHash, newQueryKeys, newProof)).resolves.toEqual(true);
```

Comment:
I guess it is not necessary, because we just want to check `verify` function and the `verifyInclusion` and `non-inclusion` already tested with same data in other unit tests.